default-fields/0000-default-fields.md

```diff
@@ -6,19 +6,224 @@
+let foo = data::Foo {
+    a: "Hello",
+    b: true,
+    c: 42
+}
+```
+
+To work around these shortcomings, users can create constructor functions or more elaborate builders.
+The problem with a constructor is that it needs to be kept in sync with the fields of the struct, or
+require the user reassign values they want to change.
+Builders enable more advanced construction, but also need to kept in-line with the struct they build,
+and can be overkill for large, but uncomplicated structs.
+
+With field defaults a caller can instantiate a struct without needing builders
+or a constructor function:
+
+```rust
+struct Foo {
+    a: &'static str,
+    b: bool = true,
+    c: i32
+}
+
+let foo = Foo {
+    a: "Hello",
+    c: 42
+}
+```
+
+This also means additional fields can be added to a struct in a non-breaking fashion.
+Say we have the following API and consumer:
+
+```rust
+mod data {
+    pub struct Foo {
+        pub a: &'static str,
+        pub c: i32
+    }
+}
+
+let foo = data::Foo {
+    a: "Hello",
+    c: 42
+}
+```
+
+We can add a new field b to this struct with a default value, and the calling code
+doesn't need to change:
+
+```rust
+mod data {
+    pub struct Foo {
+        pub a: &'static str,
+        pub b: bool = true
+        pub c: i32
+    }
+}
+
+let foo = data::Foo {
+    a: "Hello",
+    c: 42
+}
+```
+
+This is the key use-case for avoiding special opt-in syntax like `..` to utilise
+default fields.
+If it's unclear whether or not a particular caller will use a default field value then
+its addition can't be treated as a non-breaking change.
+
+The goal of this syntax is to let users build a struct as if it default fields weren't there.
 
 # Detailed design
 [design]: #detailed-design
 
-This is the bulk of the RFC. Explain the design in enough detail for somebody familiar
-with the language to understand, and for somebody familiar with the compiler to implement.
-This should get into specifics and corner-cases, and include examples of how the feature is used.
+The semantics of field defaults are the same as consts.
+The type must be supplied, and the value must be a valid compile-time constant.
+
+Field defaults take precedence over `Default::default()`.
+
+```rust
+#[derive(Default)]
+struct Foo {
+    a: &'static str,
+    b: bool = true
+    c: i32
+}
+
+// `b` is `true`, even though `bool::default()` is `false`
+let foo = Foo::default();
+```
+
+The `#[derive(Default)]` above is functionally equivalent to:
+
+```rust
+impl Default for Foo {
+	fn default() -> Self {
+		Foo {
+			a: <&'static str>::default(),
+			c: i32::default()
+		}
+	}
+}
+```
+
+Supplied field values take precedence over field defaults:
+
+```rust
+struct Foo {
+    a: &'static str,
+    b: bool = true
+    c: i32
+}
+
+// `b` is `false`, even though the field default is `true`
+let foo = Foo {
+    a: "Hello",
+    b: false,
+    c: i32
+};
+```
+
+That means if `Default` is manually implemented for a struct and default fields are
```

Comment:
I don't think you need this section on Default. On the other hand, I would add some detail on function update syntax.